searx/network/client.py

```diff
@@ -1,35 +1,58 @@
 # SPDX-License-Identifier: AGPL-3.0-or-later
 # lint: pylint
-# pylint: disable=missing-module-docstring, global-statement
+# pylint: disable=global-statement
+# pyright: basic
+"""Implement various ABCHTTPClient
+
+* OneHTTPClient        wrapper around httpx.Client
+* HTTPClient           httpx.Client accept the verify and max_redirects parameter only in the constructor.
+                       HTTPClient allows to pass these parameter in each query by creating multiple OneHTTPClient.
+* HTTPClientSoftError  Inherit from HTTPClient, raise an error according to retry_on_http_error parameter.
+* TorHTTPClient        Inherit from HTTPClientSoftError, check Tor connectivity
+"""
```

Comment:
Find better names for the classes: it is not clear that `HTTPClientSoftError` is class that should be use outside this module.